Libplanet/Blockchain/BlockChain.cs

```diff
@@ -869,10 +879,18 @@ internal void Swap(BlockChain<T> other, bool render)
                     b = Blocks[ph]
                 )
                 {
-                    var actions = b.EvaluateActionsPerTx(a =>
-                        GetStates(new[] { a }, b.PreviousHash).GetValueOrDefault(a)
-                    ).Reverse();
-                    foreach (var (_, evaluation) in actions)
+                    List<ActionEvaluation> evaluations = b.EvaluateActionsPerTx(
+                            a => GetStates(new[] { a }, b.PreviousHash).GetValueOrDefault(a))
+                        .Select(te => te.Item2).ToList();
+
+                    if (!(Policy.BlockAction is null))
```

Comment:
It would be more straight-forward:

```suggestion
                    if (Policy.BlockAction is IAction)
```